CI: Support latest GitHub runners, use latest working EDK2 release tag

olljanat opened the pull request:
I was planning to create pull request to add support for `loongarch64` but noticed that CI broken this PR fixes those:

Reasons to changes:
- Ubuntu version is updated in GitHub runners so there is small change needed to package name.
- Latest Windows SDK does not support 32-bit ARM anymore (look: https://github.com/zufuliu/notepad4/issues/839)
- EDK2 have been heavily modified so locking to last working release tag for now.

olljanat committed:
CI: Support latest GitHub runners, use latest working EDK2 release tag

pbatard commented:
Sounds good. I'm just going to move the EDK2 tag into an environment variable, so that it's easier to spot that it's something that should be updated (and make sure that we actually use the latest working EDK2 which is 2024.11 rather than 2024.08).

Note that if you plan to add LoongArch support (which I am planning to do regardless at some stage), you may want to look at what I did [in EfiFs](https://github.com/pbatard/EfiFs/blob/master/.github/workflows/linux_gcc_edk2.yml) or [in UEFI Shell](https://github.com/pbatard/UEFI-Shell/blob/main/.github/workflows/linux_gcc_edk2.yml) because I'll want something that's very close to that.

But many thanks for this PR!

pbatard commented:
And now I know why you used 2024.08... I remember having to fix the stuff that moved around, but please, in the future, avoid proposing PRs that hold back on a toolchain because it breaks compilation, and instead try to propose a fix so that compilation works with the updated toolchain.

pbatard commented:
I have now fixed the build with recent EDK2 and added LoongArch64 build support while I was at it. See a33daa5af6667a034b0eff1881df16b9734d7a8f. I'll add LoongArch64 to the tests at a later stage...

olljanat commented:
> I have now fixed the build with recent EDK2 and added LoongArch64 build support while I was at it. 

That was fast thanks.

> And now I know why you used 2024.08... I remember having to fix the stuff that moved around, but please, in the future, avoid proposing PRs that hold back on a toolchain because it breaks compilation, and instead try to propose a fix so that compilation works with the updated toolchain.

Well, I wasn’t actually able to build this locally at all and when you are first pull request creator in repo you also don’t want spent too much time before seeing that how PRs are handled.
